pkg/apis/pipeline/v1/pipeline_validation.go

```diff
@@ -91,6 +86,7 @@ func (ps *PipelineSpec) Validate(ctx context.Context) (errs *apis.FieldError) {
 	errs = errs.Also(validateWhenExpressions(ps.Tasks, ps.Finally))
 	errs = errs.Also(validateMatrix(ctx, ps.Tasks).ViaField("tasks"))
 	errs = errs.Also(validateMatrix(ctx, ps.Finally).ViaField("finally"))
+	errs = errs.Also(ps.ValidateBetaFields(ctx))
```

Comment:
NIT: should we consider this moving to the top as a first check? not that it stops any further validation but the error regarding the feature flag misconfiguration is up front.

Reply:
Thanks for your time and the review! I've updated the PR according to the pointers 🙏